server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -307,7 +307,7 @@ MappedFieldType failIfFieldMappingNotFound(String name, MappedFieldType fieldMap
     public SearchLookup lookup() {
         if (this.lookup == null) {
             this.lookup = new SearchLookup(
-                mapperService,
+                this::getFieldType,
```

Comment:
this means the lookup does a tiny bit more than before: handling unmapped fields. I wonder if that is good or bad, but tests seem to be happy either way. If we want to preserve the previous behaviour, we can easily do so.